### PR TITLE
Fixes 646 - added route for default gateway

### PR DESCRIPTION
**Pull request by syelburgi**
Created New pull request

### CODE REVIEW

File: pkg/ipam/ipam_linux.go [hunk 1]
```diff
@@ -103,6 +103,20 @@ func ConfigureIface(ifName string, res *current.Result) error {
 		ip.SettleAddresses(ifName, 10)
 	}
 
+	if v4gw != nil {
+        	route := netlink.Route{LinkIndex: link.Attrs().Index, Gw:v4gw, Flags: 0x10}
```
Comment (dcbw) [inline pkg/ipam/ipam_linux.go:107]: Looks like there's still some whitespace issues here?

**Comment by dcbw** [comment]
@syelburgi looks like there are some testcase failures, could you investigate?

`plugin type="bridge" failed (add): failed to add default route via 10.12.2.1 dev eth0': invalid argument`

File: pkg/ipam/ipam_linux.go (hunk 1 above)
Comment (dcbw) [reply pkg/ipam/ipam_linux.go:107]: @syelburgi also, I think you might need to add the `Dst` option but set to 0.0.0.0 or ::0, right?

Comment (MikeZappa87) [reply pkg/ipam/ipam_linux.go:107]: Dst should be used, it would be 0.0.0.0/0 in this case. I don't think the flags are needed here as well?

**Comment by MikeZappa87** [comment]
I am curious about your IPAM json. The routes key allows this with no code changes. I tested this with the cnitool and verified the network namespace has a default route. @syelburgi are you able to supply the inputs? 

```json
{
   "cniVersion":"1.0.0",
   "name":"mynet",
   "plugins":[
      {
         "name":"mynet",
         "type":"macvlan",
         "master":"dummy0",
         "ipam":{
            "type":"host-local",
            "ranges":[
               [
                  {
                     "subnet":"10.88.0.0/16"
                  }
               ],
               [
                  {
                     "subnet":"2001:4860:4860::/64"
                  }
               ]
            ],
            "routes":[
               {
                  "dst":"0.0.0.0/0"
               },
               {
                  "dst":"::/0"
               }
            ]
         }
      }
   ]
}
```

**Comment by squeed** [comment]
Hi there,

I'm pretty sure we don't want to synthesize a default route. There are use-cases where one doesn't want a container to have a default route. That's why we make you explicitly specify one in the host-local configuration...

**Pull request closed without merging**